src/colorizer/Dataset.ts

```diff
@@ -406,15 +405,42 @@ export default class Dataset {
     ]);
     const [outliers, tracks, times, centroids, bounds, _loadedFrame, ...featureResults] = result;
 
-    this.outliers = outliers;
-    this.trackIds = tracks;
-    this.times = times;
-    this.centroids = centroids;
-    this.bounds = bounds;
+    // TODO: Add reporting pathway for Dataset.load?
+    if (outliers.status === "rejected") {
+      console.warn("Failed to load outliers: ", outliers.reason);
+    } else {
+      this.outliers = outliers.value;
+    }
+    if (tracks.status === "rejected") {
+      console.warn("Failed to load tracks: ", tracks.reason);
+    } else {
+      this.trackIds = tracks.value;
+    }
+    if (times.status === "rejected") {
+      throw new Error("Time data could not be loaded. Is the dataset manifest file valid?");
```

Comment:
Just curious, why didn't we include `reason` in this error message for `times`?

Reply:
Ah, including it as a console warning! The error that's shown here would be directly shown on the UI.